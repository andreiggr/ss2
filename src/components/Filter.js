import React from "react";

const Filter = ({ title, options }) => {
  return (
    <div
      style={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center"
      }}
    >
      <p style={{ marginRight: "15px" }}>{title}</p>
      <select>
        {options.map((title, i) => (
          <option key={i} value={title}>
            {title}
          </option>
        ))}
      </select>
    </div>
  );
};

export default Filter;
