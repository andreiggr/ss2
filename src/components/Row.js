import React from "react";

const Row = ({ row }) => {
  return (
    <div style={{display:"flex"}}>
      {row.map((item, i) => (
        <div
          style={{
            width: "150px",
            overflowWrap: "break-word",
            wordWrap: "break-word",
            border: "1px solid black",
            padding: "5px 5px 25px 5px"
          }}
          key={i}
        >
          <p>{item}</p>
        </div>
      ))}
    </div>
  );
};

export default Row;
