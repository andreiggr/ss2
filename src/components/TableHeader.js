import React from "react";

const TableHeader = ({ headerData }) => {
  return (
    <div style={{ display: "flex" }}>
      {headerData &&
        headerData.map((item, i) => (
          <div
            style={{
              width: "150px",
              overflowWrap: "break-word",
              wordWrap: "break-word",
              border: "1px solid black",
              padding: "5px 5px 25px 5px",
              background: "#EEEFF5"
              
            }}
            key={i}
          >
            <p style={{fontWeight:"bold"}}>{item.$t}</p>
          </div>
        ))}
    </div>
  );
};

export default TableHeader;
