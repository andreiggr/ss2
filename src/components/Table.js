import React from "react";
import Row from "./Row";
import TableHeader from "./TableHeader";
import { connect } from "react-redux";
import useFilter from "../utils/useFilter";

export const Table = ({ items, filter }) => {
  //select data for table header
  const tableHeader = items[0];

  //select data for the main table with filters applied
  const tableData = [...items.slice(1)];
  const tableFiltered = useFilter(tableData, filter)

  return (
    <div style={{ display: "flex", flexDirection: "column" }}>
      <TableHeader headerData={tableHeader} />
      {tableFiltered.map((row, index) => (
        <Row key={index} row={row} />
      ))}
      <p> Total results: {tableFiltered.length} </p>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    items: state.items,
    filter: state.filter
  };
};

export default connect(mapStateToProps)(Table);
