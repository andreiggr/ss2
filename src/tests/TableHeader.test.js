import React from "react";
import ReactDOM from "react-dom";
import { TableHeader } from "../components/TableHeader";
import { shallow } from "enzyme";

describe("TableHeader Render Test", () => {
  it("correctly display header data", () => {
    const headerData = [
      "test1",
      "test2",
      "test3",
      "test4",
      "test5",
      "test6",
      "test7"
    ];
    const appComponent = shallow(<TableHeader headerData={headerData} />);

    expect(appComponent.getElements()).toMatchSnapshot();
  });
});
