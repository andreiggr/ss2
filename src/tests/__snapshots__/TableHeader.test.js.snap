exports[`TableHeader Render Test correctly display header data 1`] = `
Array [
  <div
    style={
        Object {
            "display": "flex",
          }
    }>
    <div
        style={
            Object {
                "background": "#EEEFF5",
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p
            style={
                Object {
                    "fontWeight": "bold",
                  }
            }>
            test1
        </p>
    </div>
    <div
        style={
            Object {
                "background": "#EEEFF5",
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p
            style={
                Object {
                    "fontWeight": "bold",
                  }
            }>
            test2
        </p>
    </div>
    <div
        style={
            Object {
                "background": "#EEEFF5",
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p
            style={
                Object {
                    "fontWeight": "bold",
                  }
            }>
            test3
        </p>
    </div>
    <div
        style={
            Object {
                "background": "#EEEFF5",
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p
            style={
                Object {
                    "fontWeight": "bold",
                  }
            }>
            test4
        </p>
    </div>
    <div
        style={
            Object {
                "background": "#EEEFF5",
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p
            style={
                Object {
                    "fontWeight": "bold",
                  }
            }>
            test5
        </p>
    </div>
    <div
        style={
            Object {
                "background": "#EEEFF5",
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p
            style={
                Object {
                    "fontWeight": "bold",
                  }
            }>
            test6
        </p>
    </div>
    <div
        style={
            Object {
                "background": "#EEEFF5",
                "border": "1px solid black",
                "overflowWrap": "break-word",
                "padding": "5px 5px 25px 5px",
                "width": "150px",
                "wordWrap": "break-word",
              }
        }>
        <p
            style={
                Object {
                    "fontWeight": "bold",
                  }
            }>
            test7
        </p>
    </div>
</div>,
]
`;
